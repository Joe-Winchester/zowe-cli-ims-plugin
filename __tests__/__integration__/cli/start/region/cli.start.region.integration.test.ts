import { TestEnvironment } from "../../../../__src__/environment/TestEnvironment";
import { ITestEnvironment } from "../../../../__src__/environment/doc/response/ITestEnvironment";
import { runCliScript } from "../../../../__src__/TestUtils";
import { join } from "path";

let testEnvironment: ITestEnvironment;
describe("Start region command", () => {

    beforeAll(async () => {
        testEnvironment = await TestEnvironment.setUp({
            testName: "start_region_cli_integration",
            installPlugin: true
        });
    });

    afterAll(async () => {
        await TestEnvironment.cleanUp(testEnvironment);
    });

    it("Should start a region by specifying a member name", async () => {
        const result = runCliScript(join(__dirname, "__scripts__", "start_region_help.sh"), testEnvironment);
        expect(result.stderr.toString()).toEqual("");
        expect(result.status).toEqual(0);
        expect(result.stdout.toString()).toMatchSnapshot();
    });

});
