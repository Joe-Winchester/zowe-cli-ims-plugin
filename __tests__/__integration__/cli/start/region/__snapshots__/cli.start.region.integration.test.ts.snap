// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start region command Should start a region by specifying a member name 1`] = `
"
 COMMAND NAME
 ------------

   region | reg

 DESCRIPTION
 -----------

   Start an IMS region.

 USAGE
 -----

   zowe ims start region [memberName] [options]

 POSITIONAL ARGUMENTS
 --------------------

   memberName		 (string)

      The name of the member containing JCL for the region to start. The maximum
      length of the member name is eight characters. If no member name is specified,
      the default member name is used

 OPTIONS
 -------

   --local  | -l (boolean)

      If you specify the --local option, IMS overrides the symbolic IMSID parameter in
      the JCL of the default or specified member. --local is the default if you
      specify the --job-name option.

      Default value: ALL

   --job-name  | --jn (string)

      Use this option to override the job name on the JOB statement of the default or
      specified JCL member for a dependent region

 IMS CONNECTION OPTIONS
 ----------------------

   --host  | -H (string)

      The IMS Command Services server host name.

   --port  | -P (number)

      The IMS Command Services server port.

   --ims-connect-host  | --ich (string)

      The hostname of your instance of IMS Connect. This is typically the hostname of
      the mainframe LPAR where IMS Connect is running.

   --ims-connect-port  | --icp (number)

      The port of your instance of IMS Connect. This port can be found in your IMS
      Connect configuration file on the mainframe.

   --plex  | -x (string)

      The name of the IMS plex.

   --user  | -u (string)

      Mainframe (IMS) user name, which can be the same as your TSO login.

   --password  | --pass (string)

      Mainframe (IMS) password, which can be the same as your TSO password.

 PROFILE OPTIONS
 ---------------

   --ims-profile  | --ims-p (string)

      The name of a (ims) profile to load for this command execution.

 RESPONSE FORMAT OPTIONS
 -----------------------

   --response-format-filter  | --rff (array)

      Filter (include) fields in the response. Accepts an array of field/property
      names to include in the output response. You can filter JSON objects properties
      OR table columns/fields. In addition, you can use this option in conjunction
      with '--response-format-type' to reduce the output of a command to a single
      field/property or a list of a single field/property.

   --response-format-type  | --rft (string)

      The command response output format type. Must be one of the following:

      table: Formats output data as a table. Use this option when the output data is
      an array of homogeneous JSON objects. Each property of the object will become a
      column in the table.

      list: Formats output data as a list of strings. Can be used on any data type
      (JSON objects/arrays) are stringified and a new line is added after each entry
      in an array.

      object: Formats output data as a list of prettified objects (or single object).
      Can be used in place of \\"table\\" to change from tabular output to a list of
      prettified objects.

      string: Formats output data as a string. JSON objects/arrays are stringified.

      Allowed values: table, list, object, string

   --response-format-header  | --rfh (boolean)

      If \\"--response-format-type table\\" is specified, include the column headers in
      the output.

 GLOBAL OPTIONS
 --------------

   --response-format-json  | --rfj (boolean)

      Produce JSON formatted data from a command

   --help  | -h (boolean)

      Display help text


 EXAMPLES
 --------

   - Start a region stored in a member named MEM1:

      $ zowe ims start region MEM1

{
  \\"success\\": true,
  \\"message\\": \\"The help was constructed for command: region.\\",
  \\"stdout\\": \\"\\\\n COMMAND NAME\\\\n ------------\\\\n\\\\n   region | reg\\\\n\\\\n DESCRIPTION\\\\n -----------\\\\n\\\\n   Start an IMS region.\\\\n\\\\n USAGE\\\\n -----\\\\n\\\\n   zowe ims start region [memberName] [options]\\\\n\\\\n POSITIONAL ARGUMENTS\\\\n --------------------\\\\n\\\\n   memberName\\\\t\\\\t (string)\\\\n\\\\n      The name of the member containing JCL for the region to start. The maximum\\\\n      length of the member name is eight characters. If no member name is specified,\\\\n      the default member name is used\\\\n\\\\n OPTIONS\\\\n -------\\\\n\\\\n   --local  | -l (boolean)\\\\n\\\\n      If you specify the --local option, IMS overrides the symbolic IMSID parameter in\\\\n      the JCL of the default or specified member. --local is the default if you\\\\n      specify the --job-name option.\\\\n\\\\n      Default value: ALL\\\\n\\\\n   --job-name  | --jn (string)\\\\n\\\\n      Use this option to override the job name on the JOB statement of the default or\\\\n      specified JCL member for a dependent region\\\\n\\\\n IMS CONNECTION OPTIONS\\\\n ----------------------\\\\n\\\\n   --host  | -H (string)\\\\n\\\\n      The IMS Command Services server host name.\\\\n\\\\n   --port  | -P (number)\\\\n\\\\n      The IMS Command Services server port.\\\\n\\\\n   --ims-connect-host  | --ich (string)\\\\n\\\\n      The hostname of your instance of IMS Connect. This is typically the hostname of\\\\n      the mainframe LPAR where IMS Connect is running.\\\\n\\\\n   --ims-connect-port  | --icp (number)\\\\n\\\\n      The port of your instance of IMS Connect. This port can be found in your IMS\\\\n      Connect configuration file on the mainframe.\\\\n\\\\n   --plex  | -x (string)\\\\n\\\\n      The name of the IMS plex.\\\\n\\\\n   --user  | -u (string)\\\\n\\\\n      Mainframe (IMS) user name, which can be the same as your TSO login.\\\\n\\\\n   --password  | --pass (string)\\\\n\\\\n      Mainframe (IMS) password, which can be the same as your TSO password.\\\\n\\\\n PROFILE OPTIONS\\\\n ---------------\\\\n\\\\n   --ims-profile  | --ims-p (string)\\\\n\\\\n      The name of a (ims) profile to load for this command execution.\\\\n\\\\n RESPONSE FORMAT OPTIONS\\\\n -----------------------\\\\n\\\\n   --response-format-filter  | --rff (array)\\\\n\\\\n      Filter (include) fields in the response. Accepts an array of field/property\\\\n      names to include in the output response. You can filter JSON objects properties\\\\n      OR table columns/fields. In addition, you can use this option in conjunction\\\\n      with '--response-format-type' to reduce the output of a command to a single\\\\n      field/property or a list of a single field/property.\\\\n\\\\n   --response-format-type  | --rft (string)\\\\n\\\\n      The command response output format type. Must be one of the following:\\\\n\\\\n      table: Formats output data as a table. Use this option when the output data is\\\\n      an array of homogeneous JSON objects. Each property of the object will become a\\\\n      column in the table.\\\\n\\\\n      list: Formats output data as a list of strings. Can be used on any data type\\\\n      (JSON objects/arrays) are stringified and a new line is added after each entry\\\\n      in an array.\\\\n\\\\n      object: Formats output data as a list of prettified objects (or single object).\\\\n      Can be used in place of \\\\\\"table\\\\\\" to change from tabular output to a list of\\\\n      prettified objects.\\\\n\\\\n      string: Formats output data as a string. JSON objects/arrays are stringified.\\\\n\\\\n      Allowed values: table, list, object, string\\\\n\\\\n   --response-format-header  | --rfh (boolean)\\\\n\\\\n      If \\\\\\"--response-format-type table\\\\\\" is specified, include the column headers in\\\\n      the output.\\\\n\\\\n GLOBAL OPTIONS\\\\n --------------\\\\n\\\\n   --response-format-json  | --rfj (boolean)\\\\n\\\\n      Produce JSON formatted data from a command\\\\n\\\\n   --help  | -h (boolean)\\\\n\\\\n      Display help text\\\\n\\\\n\\\\n EXAMPLES\\\\n --------\\\\n\\\\n   - Start a region stored in a member named MEM1:\\\\n\\\\n      $ zowe ims start region MEM1\\\\n\\\\n\\",
  \\"stderr\\": \\"\\",
  \\"data\\": \\"\\\\n COMMAND NAME\\\\n ------------\\\\n\\\\n   region | reg\\\\n\\\\n DESCRIPTION\\\\n -----------\\\\n\\\\n   Start an IMS region.\\\\n\\\\n USAGE\\\\n -----\\\\n\\\\n   zowe ims start region [memberName] [options]\\\\n\\\\n POSITIONAL ARGUMENTS\\\\n --------------------\\\\n\\\\n   memberName\\\\t\\\\t (string)\\\\n\\\\n      The name of the member containing JCL for the region to start. The maximum\\\\n      length of the member name is eight characters. If no member name is specified,\\\\n      the default member name is used\\\\n\\\\n OPTIONS\\\\n -------\\\\n\\\\n   --local  | -l (boolean)\\\\n\\\\n      If you specify the --local option, IMS overrides the symbolic IMSID parameter in\\\\n      the JCL of the default or specified member. --local is the default if you\\\\n      specify the --job-name option.\\\\n\\\\n      Default value: ALL\\\\n\\\\n   --job-name  | --jn (string)\\\\n\\\\n      Use this option to override the job name on the JOB statement of the default or\\\\n      specified JCL member for a dependent region\\\\n\\\\n IMS CONNECTION OPTIONS\\\\n ----------------------\\\\n\\\\n   --host  | -H (string)\\\\n\\\\n      The IMS Command Services server host name.\\\\n\\\\n   --port  | -P (number)\\\\n\\\\n      The IMS Command Services server port.\\\\n\\\\n   --ims-connect-host  | --ich (string)\\\\n\\\\n      The hostname of your instance of IMS Connect. This is typically the hostname of\\\\n      the mainframe LPAR where IMS Connect is running.\\\\n\\\\n   --ims-connect-port  | --icp (number)\\\\n\\\\n      The port of your instance of IMS Connect. This port can be found in your IMS\\\\n      Connect configuration file on the mainframe.\\\\n\\\\n   --plex  | -x (string)\\\\n\\\\n      The name of the IMS plex.\\\\n\\\\n   --user  | -u (string)\\\\n\\\\n      Mainframe (IMS) user name, which can be the same as your TSO login.\\\\n\\\\n   --password  | --pass (string)\\\\n\\\\n      Mainframe (IMS) password, which can be the same as your TSO password.\\\\n\\\\n PROFILE OPTIONS\\\\n ---------------\\\\n\\\\n   --ims-profile  | --ims-p (string)\\\\n\\\\n      The name of a (ims) profile to load for this command execution.\\\\n\\\\n RESPONSE FORMAT OPTIONS\\\\n -----------------------\\\\n\\\\n   --response-format-filter  | --rff (array)\\\\n\\\\n      Filter (include) fields in the response. Accepts an array of field/property\\\\n      names to include in the output response. You can filter JSON objects properties\\\\n      OR table columns/fields. In addition, you can use this option in conjunction\\\\n      with '--response-format-type' to reduce the output of a command to a single\\\\n      field/property or a list of a single field/property.\\\\n\\\\n   --response-format-type  | --rft (string)\\\\n\\\\n      The command response output format type. Must be one of the following:\\\\n\\\\n      table: Formats output data as a table. Use this option when the output data is\\\\n      an array of homogeneous JSON objects. Each property of the object will become a\\\\n      column in the table.\\\\n\\\\n      list: Formats output data as a list of strings. Can be used on any data type\\\\n      (JSON objects/arrays) are stringified and a new line is added after each entry\\\\n      in an array.\\\\n\\\\n      object: Formats output data as a list of prettified objects (or single object).\\\\n      Can be used in place of \\\\\\"table\\\\\\" to change from tabular output to a list of\\\\n      prettified objects.\\\\n\\\\n      string: Formats output data as a string. JSON objects/arrays are stringified.\\\\n\\\\n      Allowed values: table, list, object, string\\\\n\\\\n   --response-format-header  | --rfh (boolean)\\\\n\\\\n      If \\\\\\"--response-format-type table\\\\\\" is specified, include the column headers in\\\\n      the output.\\\\n\\\\n GLOBAL OPTIONS\\\\n --------------\\\\n\\\\n   --response-format-json  | --rfj (boolean)\\\\n\\\\n      Produce JSON formatted data from a command\\\\n\\\\n   --help  | -h (boolean)\\\\n\\\\n      Display help text\\\\n\\\\n\\\\n EXAMPLES\\\\n --------\\\\n\\\\n   - Start a region stored in a member named MEM1:\\\\n\\\\n      $ zowe ims start region MEM1\\\\n\\\\n\\"
}"
`;
